### ISSUE TITLE
StackOverflowError

### ISSUE THREAD

**Issue by klaus7** [mariodavid/jmix-sneferu#1]
I wanted to try it out, but everytime I add the the dependency (`testImplementation 'de.diedavids.jmix.sneferu:jmix-sneferu:1.0.0'`) to my build.gradle, IntelliJ hangs and I can't open my files anymore. 

Sneferu can't be found in the stacktrace, so maybe it's a jmix or even a intellij problem.

**IntelliJ**
IntelliJ IDEA 2021.3.2 (Ultimate Edition)
Build #IU-213.6777.52, built on January 28, 2022
Licensed to Klaus Pfeiffer
Subscription is active until April 9, 2022.
Runtime version: 11.0.13+7-b1751.25 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.13.0-10027-tuxedo
GC: G1 Young Generation, G1 Old Generation
Memory: 8200M
Cores: 16
Non-Bundled Plugins:
    com.intellij.plugins.watcher (213.6461.19)
    org.jetbrains.kotlin (213-1.6.10-release-961-IJ6777.52)
    VisualVMLauncher (1.20.203.4204.0)
    String Manipulation (9.2.0)
    mobi.hsz.idea.gitignore (4.3.0)
    com.jetbrains.php (213.6777.58)
    io.jmix.studio (1.1.6-213)
    CUBA (15.8-213)

Kotlin: 213-1.6.10-release-961-IJ6777.52
Current Desktop: ubuntu:GNOME

Jmix Plugin: 1.1.6-213

**Stacktrace:**
```
java.lang.StackOverflowError
	at com.intellij.psi.stubs.LazyStubList.get(LazyStubList.java:18)
	at com.intellij.psi.stubs.StubList$1.get(StubList.java:155)
	at com.intellij.psi.stubs.StubList$1.get(StubList.java:151)
	at com.intellij.psi.stubs.StubBase.fillFilteredChildren(StubBase.java:129)
	at com.intellij.psi.stubs.StubBase.getChildrenByType(StubBase.java:98)
	at com.intellij.psi.impl.compiled.ClsClassImpl.getOwnInnerClasses(ClsClassImpl.java:193)
	at com.intellij.psi.impl.compiled.ClsClassImpl.getChildren(ClsClassImpl.java:52)
	at com.intellij.psi.util.PsiTreeUtil.processElements(PsiTreeUtil.java:913)
	at com.intellij.psi.util.PsiTreeUtil.findChildrenOfAnyType(PsiTreeUtil.java:316)
	at com.intellij.psi.util.PsiTreeUtil.findChildrenOfType(PsiTreeUtil.java:281)
	at com.intellij.psi.impl.search.AnnotatedElementsSearcher.lambda$getAnnotationCandidates$3(AnnotatedElementsSearcher.java:77)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:865)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.psi.impl.search.AnnotatedElementsSearcher.getAnnotationCandidates(AnnotatedElementsSearcher.java:70)
	at com.intellij.psi.impl.search.AnnotatedElementsSearcher.execute(AnnotatedElementsSearcher.java:35)
	at com.intellij.psi.impl.search.AnnotatedElementsSearcher.execute(AnnotatedElementsSearcher.java:21)
	at com.intellij.util.ExecutorsQuery.processResults(ExecutorsQuery.java:28)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:91)
	at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:108)
	at com.intellij.util.InstanceofQuery.processResults(InstanceofQuery.java:22)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:91)
	at com.intellij.util.AbstractQuery.forEach(AbstractQuery.java:83)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentSearch.lambda$walkAnnotatedMethods$10(StudioComponentSearch.java:328)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:839)
	at com.haulmont.jmixstudio.util.EntityUtil.ra(EntityUtil.java:859)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentSearch.walkAnnotatedMethods(StudioComponentSearch.java:326)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentSearch.getStudioAnnotatedMethods(StudioComponentSearch.java:303)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.collectMethodBasedProperties(StudioPropertiesCreator.java:249)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.lambda$collectMethodBasedProperties$9(StudioPropertiesCreator.java:258)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at one.util.streamex.AbstractStreamEx.forEach(AbstractStreamEx.java:359)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.collectMethodBasedProperties(StudioPropertiesCreator.java:258)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.lambda$collectMethodBasedProperties$9(StudioPropertiesCreator.java:258)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at one.util.streamex.AbstractStreamEx.forEach(AbstractStreamEx.java:359)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.collectMethodBasedProperties(StudioPropertiesCreator.java:258)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.lambda$collectMethodBasedProperties$9(StudioPropertiesCreator.java:258)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at one.util.streamex.AbstractStreamEx.forEach(AbstractStreamEx.java:359)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.collectMethodBasedProperties(StudioPropertiesCreator.java:258)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioPropertiesCreator.getStudioComponentProperties(StudioPropertiesCreator.java:81)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:807)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.initComponentContainer(StudioComponentLibrary.kt:789)
	at com.haulmont.jmixstudio.intellij.ui.sd.designer.meta.StudioComponentLibrary.createElement(StudioComponentLibrary.kt:720)
...
```

**Comment by mariodavid** [comment]
Hi @klaus7 ,

not sure where this is coming from TBH.

Attached I created a dedicated sample application using the 1.2-RC1 of Jmix (I had to exclude the transitive dependencies from Sneferu since they are still pointing to 1.2-SNAPSHOT, this will not be needed anymore when 1.2 is out and I have published a 1.1 of Sneferu).

[jmix-sneferu-example.zip](https://github.com/mariodavid/jmix-sneferu/files/8139413/jmix-sneferu-example.zip)

Can you take a look if this causes the same problem? Locally for me, this worked...

**Comment by klaus7** [comment]
Hi @mariodavid , thanks for your fast response. As 1.2 RC1 was announced yesterday I decided to wait for the stable release and then give it another try. I will report back to you.

**Comment by mariodavid** [comment]
hi @klaus7, I recently released 1.1 with support for Jmix 1.2. With that everything should work fine (I hope :))


Cheers
Mario

**Comment by klaus7** [comment]
StackOverflowError is fixed it seems. :) I wrote my first test, but encountered another problem.
